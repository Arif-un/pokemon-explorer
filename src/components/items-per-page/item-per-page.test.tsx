import { fireEvent, render, screen } from '@testing-library/react'
import { describe, expect, it, vi } from 'vitest'

import ItemsPerPage from './items-per-page'

describe('ItemsPerPage Component', () => {
  it('renders correctly and matches the snapshot', () => {
    const { asFragment } = render(<ItemsPerPage itemsPerPage={20} onItemsPerPageChange={() => void 0} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('calls onItemsPerPageChange with the correct value when a new option is selected', () => {
    const mockOnItemsPerPageChange = vi.fn()
    render(<ItemsPerPage itemsPerPage={20} onItemsPerPageChange={mockOnItemsPerPageChange} />)

    const selectElement = screen.getByRole('combobox')

    fireEvent.change(selectElement, { target: { value: '50' } })

    expect(mockOnItemsPerPageChange).toHaveBeenCalledWith(50)
  })

  it('renders with the correct initial value', () => {
    render(<ItemsPerPage itemsPerPage={50} onItemsPerPageChange={() => void 0} />)
    const selectElement = screen.getByRole('combobox')
    expect(selectElement).toHaveValue('50')
  })
})
