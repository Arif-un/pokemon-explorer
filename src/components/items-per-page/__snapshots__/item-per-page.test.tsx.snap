// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ItemsPerPage Component > renders correctly and matches the snapshot 1`] = `
<DocumentFragment>
  <label
    class="flex items-center gap-3 text-slate-500"
  >
    <span>
      Showing
    </span>
    <select
      class="p-2 rounded-full cursor-pointer border border-slate-200 bg-slate-50 text-slate-800 focus:outline-none focus-visible:ring-2 ring-yellow-500  dark:bg-slate-900 dark:text-slate-100 dark:border-slate-800"
      name="items-per-page"
    >
      <option
        value="20"
      >
        20
      </option>
      <option
        value="50"
      >
        50
      </option>
      <option
        value="100"
      >
        100
      </option>
    </select>
    <span>
      items per page
    </span>
  </label>
</DocumentFragment>
`;
